### PR TITLE
Update porter in RP and remove support for v0

### LINKED ISSUE #2990: Remove Porter v0
## Description

As a TRE Developer
I want to remove Porter v0 after v1 has been deployed for a while.

## Acceptance criteria

- [ ] Devcontainer doesn't contain v0
- [ ] RP image doesn't contain v0
- [ ] https://github.com/microsoft/AzureTRE/blob/main/resource_processor/run.sh has migration code removed

### LINKED ISSUE #4224: Sync Porter Updates in Resource Processor
## Description

As a TRE Developer
I want to Porter (and components) used by the resource processor to be the same as in the dev container
So that a cloud deployment will function as expected by the developer

## Acceptance criteria

- [ ] Versions in the resource processor match the ones in the dev container

### PR DESCRIPTION
**Pull request by tamirkamara**
# Resolves #4224, Resolves #2990 

## What is being addressed

Porter used in the Resource Processor need to be synced with what we use in the dev container.

## How is this addressed

- Update RP to use the same version of Porter as in the dev container
- Remove support for Porter v0

### CODE REVIEW

File: templates/shared_services/certs/terraform/staticweb.tf [hunk 1]
```diff
@@ -8,7 +8,7 @@ resource "azurerm_storage_account" "staticweb" {
   account_replication_type         = "LRS"
   table_encryption_key_type        = var.enable_cmk_encryption ? "Account" : "Service"
   queue_encryption_key_type        = var.enable_cmk_encryption ? "Account" : "Service"
-  enable_https_traffic_only        = true
+  https_traffic_only_enabled       = true
```
Comment (jonnyry) [inline templates/shared_services/certs/terraform/staticweb.tf:11]: Is this intentionally part of this change?

File: core/terraform/appgateway/staticweb.tf [hunk 2]
```diff
@@ -8,7 +8,7 @@ resource "azurerm_storage_account" "staticweb" {
   account_replication_type         = "LRS"
   table_encryption_key_type        = var.enable_cmk_encryption ? "Account" : "Service"
   queue_encryption_key_type        = var.enable_cmk_encryption ? "Account" : "Service"
-  enable_https_traffic_only        = true
+  https_traffic_only_enabled       = true
```
Comment (jonnyry) [inline core/terraform/appgateway/staticweb.tf:11]: is this intentionally part of this change?

File: templates/shared_services/certs/terraform/staticweb.tf (hunk 1 above)
Comment (tamirkamara) [reply templates/shared_services/certs/terraform/staticweb.tf:11]: Yes. I know it's not related to this PR but is a small thing that will remove an annoying warning. I can remove if needed.

File: core/terraform/appgateway/staticweb.tf (hunk 2 above)
Comment (tamirkamara) [reply core/terraform/appgateway/staticweb.tf:11]: Yes. I know it's not related to this PR but is a small thing that will remove an annoying warning. I can remove if needed.

Comment (jonnyry) [reply core/terraform/appgateway/staticweb.tf:11]: OK thats fine to leave in, just checking :-)

File: templates/shared_services/certs/terraform/staticweb.tf (hunk 1 above)
Comment (jonnyry) [reply templates/shared_services/certs/terraform/staticweb.tf:11]: OK thats fine to leave in, just checking :-)

**Review by jonnyry — APPROVED** [review]
Core requires another version bump after the merge.  Other than that LGTM.

**Comment by tamirkamara** [comment]
/test-force-approve
passing tests here: https://github.com/microsoft/AzureTRE/actions/runs/12550445692

**Review by LizaShak — APPROVED** [review]

**Comment by tamirkamara** [comment]
/test-force-approve
passing tests here: https://github.com/microsoft/AzureTRE/actions/runs/12550445692

**Pull request merged**